astropy/io/fits/column.py

```diff
@@ -1429,6 +1429,12 @@ def dtype(self):
                 else:
                     dt = np.dtype((dt.base, dim))
 
+            # On Python 2, force the `name` to `str`
+            # because Numpy structured arrays can't handle unicode `name`
+            # See https://github.com/astropy/astropy/issues/5204
+            if PY2:  # pragma: py2
+                name = str(name)
```

Comment:
The error message if someone tries to put non-ascii characters for a FITS Column name is already a `ValueError` with a very good error message:
```python
>>> from astropy.io import fits
>>> fits.Column(name=u'Grüffelo', format='E', array=[42.])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astropy/io/fits/column.py", line 552, in __init__
    setattr(self, attr, valid_kwargs.get(attr))
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astropy/io/fits/column.py", line 444, in __set__
    self._validator(obj, value)
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astropy/io/fits/column.py", line 768, in name
    len(str(Card('TTYPE', name))) == CARD_LENGTH), \
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astropy/io/fits/card.py", line 448, in __init__
    self.value = value
  File "/opt/local/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/astropy/io/fits/card.py", line 582, in value
    'ASCII or non-printable characters.' % value)
ValueError: FITS header values must contain standard printable ASCII characters; u'Gr\xfcffelo' contains characters not representable in ASCII or non-printable characters.
```
So I think it's OK.